reference/test/matrix/permutation.cpp

```diff
@@ -67,413 +68,97 @@ TYPED_TEST_SUITE(Permutation, gko::test::ValueIndexTypes,
-    // clang-format off
-    GKO_ASSERT_MTX_NEAR(y,
-                        l({{4.0, 2.5},
-                           {2.0, 3.0}}),
-                        0.0);
-    // clang-format on
+    EXPECT_EQ(inv->get_const_permutation()[0], 2);
+    EXPECT_EQ(inv->get_const_permutation()[1], 0);
+    EXPECT_EQ(inv->get_const_permutation()[2], 1);
 }
 
 
-TYPED_TEST(Permutation, AppliesColPermutationToDense)
+TYPED_TEST(Permutation, Combine)
 {
-    using i_type = typename TestFixture::i_type;
-    using T = typename TestFixture::v_type;
-    using Vec = typename TestFixture::Vec;
-    // clang-format off
-    auto x = gko::initialize<Vec>(
-        {I<T>{2.0, 3.0},
-         I<T>{4.0, 2.5}}, this->exec);
-    // clang-format on
-    auto y = Vec::create(this->exec, gko::dim<2>{2});
-    i_type rdata[] = {1, 0};
+    using index_type = typename TestFixture::index_type;
+    auto perm = gko::matrix::Permutation<index_type>::create(
+        this->exec, gko::array<index_type>{this->exec, {1, 2, 0}});
+    auto perm2 = gko::matrix::Permutation<index_type>::create(
+        this->exec, gko::array<index_type>{this->exec, {0, 2, 1}});
 
-    auto perm = gko::matrix::Permutation<i_type>::create(
-        this->exec, gko::dim<2>{2}, gko::make_array_view(this->exec, 2, rdata),
-        gko::matrix::column_permute);
+    auto combined = perm->combine(perm2);
 
-    perm->apply(x, y);
-    // clang-format off
-    GKO_ASSERT_MTX_NEAR(y,
-                        l({{3.0, 2.0},
-                           {2.5, 4.0}}),
-                        0.0);
-    // clang-format on
+    EXPECT_EQ(combined->get_const_permutation()[0], 2);
+    EXPECT_EQ(combined->get_const_permutation()[1], 1);
+    EXPECT_EQ(combined->get_const_permutation()[2], 0);
 }
 
 
-TYPED_TEST(Permutation, AppliesRowAndColPermutationToDense)
+TYPED_TEST(Permutation, CombineWithInverse)
 {
-    using i_type = typename TestFixture::i_type;
-    using T = typename TestFixture::v_type;
-    using Vec = typename TestFixture::Vec;
-    // clang-format off
-    auto x = gko::initialize<Vec>(
-        {I<T>{2.0, 3.0},
-         I<T>{4.0, 2.5}}, this->exec);
-    // clang-format on
-    auto y1 = Vec::create(this->exec, gko::dim<2>{2});
-    auto y2 = Vec::create(this->exec, gko::dim<2>{2});
-    i_type cdata[] = {1, 0};
-    i_type rdata[] = {1, 0};
-
-    auto rperm = gko::matrix::Permutation<i_type>::create(
-        this->exec, gko::dim<2>{2}, gko::make_array_view(this->exec, 2, rdata));
-    auto cperm = gko::matrix::Permutation<i_type>::create(
-        this->exec, gko::dim<2>{2}, gko::make_array_view(this->exec, 2, cdata),
-        gko::matrix::column_permute);
-
-    rperm->apply(x, y1);
-    cperm->apply(y1, y2);
-    // clang-format off
-    GKO_ASSERT_MTX_NEAR(y2,
-                        l({{2.5, 4.0},
-                           {3.0, 2.0}}),
-                        0.0);
-    // clang-format on
-}
-
-
-TYPED_TEST(Permutation, AppliesRowAndColPermutationToDenseWithOneArray)
```

Comment:
missing deleted function

Reply:
The functionality is no longer available through `Permutation`